src/libraries/System.Runtime.Caching/src/System/Runtime/Caching/MemoryCache.cs

```diff
@@ -39,8 +39,6 @@ public class MemoryCache : ObjectCache, IEnumerable, IDisposable
         private bool _useMemoryCacheManager = true;
         private bool _throwOnDisposed;
         private EventHandler _onAppDomainUnload;
```

Comment:
Should the AppDomainUnload event handler be deleted for the same reason?

Reply:
AppDomainUnload doesn't run the same danger of being called unexpectedly on a stack that might be holding locks or be in some other inconsistent state. So I don't think it's a danger here. Since System.Runtime.Caching is simply a bridge package to allow folks to migrate from NetFx to .Net, I tend to approach with a servicing mentality and only make the changes that are necessary. (Also, the 1st party customer affected by this is still targeting 6.0 with some of their services, so this actually is servicing as we will want to backport whatever we do here.)

I can add another commit if you guys disagree with the conservative approach.

Reply:
We have constant trickle of shutdown bugs (hangs or crashes) caused by unnecessary code running during shutdown. I think it is just a matter of time until somebody hits a problem with the AppDomainUnload handler.

> this actually is servicing as we will want to backport whatever we do here.

It is fine to do a minimal backport that just fixes the customer issue for servicing and do more complete fix in main. It is often how the fixes for the shutdown bugs look like. For example:
- https://github.com/dotnet/runtime/pull/100293 - full fix in main
- https://github.com/dotnet/runtime/pull/100836 - partial backport for servicing